common/types.go

```diff
@@ -371,20 +660,146 @@ func (a *Address) Scan(src interface{}) error {
 }
 
 // Value implements valuer for database/sql.
-func (a Address) Value() (driver.Value, error) {
+func (a InternalAddress) Value() (driver.Value, error) {
 	return a[:], nil
 }
 
 // IsInChainScope checks if an address is a valid account in our node's sharded address space
-func (a Address) IsInChainScope() bool {
-	if a == ZeroAddr {
+func (a InternalAddress) IsInChainScope() bool { // Eventually this will just return true
```

Comment:
This method should not exist. An InternalAddress by definition is in chain scope. It should be impossible to construct an internal address that is not in chain scope, and _if_ it was possible, it would invalidate all of the code that depends on type assertions. i.e. anywhere that checks `if _, ok := addr.(*InternalAddress); !ok {...}` would be invalid, since it would then be possible to have an `InternalAddress` whose bytes aren't actually represent an `ExternalAddress`.

I did a quick scan, and I believe everywhere you use `IsInChainScope()` could be changed to either perform a type assertion. Unless I am missing something, could you remove this method and use type assertions instead? Forcing usage of type assertions was the key objective of this PR